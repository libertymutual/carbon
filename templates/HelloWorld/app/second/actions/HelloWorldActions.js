import * as ActionTypes from '../actions/ActionTypes';

/**
 * Hello World related action creators
 */
export default {

  /**
   * Says hello world
   * @returns {function} A function which dispatches a HELLO action
   */
  hello() {
    return function (dispatch) {
      dispatch({ type: ActionTypes.HELLO });
    };
  },

  /**
   * Says goodbye world
   * @returns {function} A function which dispatches a GOODBYE action
   */
  goodbye() {
    return function (dispatch) {
      dispatch({ type: ActionTypes.GOODBYE });
    };
  },
};
