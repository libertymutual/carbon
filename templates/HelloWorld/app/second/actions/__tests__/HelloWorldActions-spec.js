import 'react-native';
import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import HelloWorldActions from '../HelloWorldActions';

describe('HelloWorldActions tests', () => {

  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  let store;
  let dispatch;

  beforeEach(() => {
    // Sets up a new store
    store = mockStore();
    dispatch = store.dispatch;
  });

  test('hello returns a function', () => {
    var dispatcher = HelloWorldActions.hello();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('hello function dispatches a HELLO action', () => {
    var dispatcher = HelloWorldActions.hello();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });

  test('goodbye returns a function', () => {
    var dispatcher = HelloWorldActions.goodbye();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('goodbye function dispatches a GOODBYE action', () => {
    var dispatcher = HelloWorldActions.goodbye();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });
});
