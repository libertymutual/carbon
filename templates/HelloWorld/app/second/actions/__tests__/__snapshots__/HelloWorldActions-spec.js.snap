// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorldActions tests goodbye function dispatches a GOODBYE action 1`] = `
Array [
  Object {
    "type": "goodbye",
  },
]
`;

exports[`HelloWorldActions tests goodbye returns a function 1`] = `"function"`;

exports[`HelloWorldActions tests hello function dispatches a HELLO action 1`] = `
Array [
  Object {
    "type": "hello",
  },
]
`;

exports[`HelloWorldActions tests hello returns a function 1`] = `"function"`;
